tools/querytee/proxy.go

```diff
@@ -153,7 +187,15 @@ func NewProxy(cfg ProxyConfig, logger log.Logger, routes []Route, registerer pro
 			preferred = preferredIdx == idx
 		}
 
-		p.backends = append(p.backends, NewProxyBackend(name, u, cfg.BackendReadTimeout, preferred, cfg.BackendSkipTLSVerify))
+		backendCfg := cfg.parsedBackendConfig[name]
+		if backendCfg == nil {
+			backendCfg = cfg.parsedBackendConfig[strconv.Itoa(idx)]
```

Comment:
Why are we looking up the config by index as well as by name?

Reply:
so we can test. we do it for the primary selector too